client/packages/common/src/ui/components/inputs/TextInput/numeric/PositiveNumberInput.tsx

```diff
@@ -31,7 +31,11 @@ export const PositiveNumberInput = React.forwardRef(
           sx: { ...sx, '& .MuiInput-input': { textAlign: 'right' } },
         }}
         onChange={value =>
-          onChange(NumUtils.constrain(value, Math.max(0, min), max))
+          onChange(
+            value !== undefined
+              ? NumUtils.constrain(value, Math.max(1, min), max)
```

Comment:
this should be `max(0, min)`?

btw. there is the idea around to remove this PostitiveNumberInput and just have a normal NumberInput where you can set the correct range... For example, to have a range [-5,5]

Reply:
The existing comment for PositiveNumberInput is `// where Positive is n >=1` but the behaviour wasn't consistent with this, so I fixed it. See storybook for the variety of number components and the ranges allowed:

![Screenshot 2023-03-29 at 11 29 55 AM](https://user-images.githubusercontent.com/5456533/228381142-a1558a2a-67c9-4ad2-b249-5ab825202ade.png)

> there is the idea around to remove this PostitiveNumberInput and just have a normal NumberInput where you can set the correct range... For example, to have a range [-5,5]

I agree that this would be better, I did think there was way too many number components with only slight differences ("Non-negative" vs "Positive"? 😳)

Yes, a single component combining these four with just a range prop seems much nicer.
![Screenshot 2023-03-29 at 11 33 29 AM](https://user-images.githubusercontent.com/5456533/228381680-a32b2afd-6059-4308-b64c-290972d352bf.png)

Shall I make an issue for that?

Reply:
> but the behaviour wasn't consistent with this, so I fixed it

> Just noticed that you can't set a value to 0 any more, e.g. for "Pill Count (at visit start)", it has a minimum of 0 but this should include 0 according to the JSON schema (https://json-schema.org/understanding-json-schema/reference/numeric.html)...

Oh, I see, even though the component said it is supposed to be`where Positive is n >=1`, the `min` property was allowing you to set it to `0`, which my change removed. It essentially makes it the same as the "NonNegativeNumberInput" when you have `min=0`,  hence my confusion.

I'll revert it now, but yes, we really should tidy these number components up, it's a bit of a shambles.

Reply:
b58c00dd1

Reply:
> Shall I make an issue for that?

Done: #1440

Reply:
thanks!